<launch>

    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="m" type="double" value="3.8" />
    <param name="I_z" type="double" value="0.20741" />
    <param name="L_a" type="double" value="0.125" />
    <param name="L_b" type="double" value="0.125" />
    <!-- control parameters -->
    <param name="input_gain" type="double" value="1.7675" />
    <param name="air_drag_coeff" type="double" value="0.33267" />
    <param name="friction" type="double" value="0.12902" />
    <!-- tire parameters -->
    <param name="tire_model/Cf" type="double" value="6.0823" />
    <param name="tire_model/Cr" type="double" value="6.0823" />
	<param name="run_time" type="double" value="5.0" />
	<param name="step_time" type="double" value="2.0" />
	<param name="motor_input" type="double" value="0.0" />
	<param name="servo_input" type="double" value="0.0" />
	<rosparam command="load" file="$(find barc)/launch/adas.yaml" />
<!-- LIDAR -->
    <node pkg="rplidar_ros" type="rplidarNode" name="lidar">
        <param name="serial_port" value="/dev/ttyUSB1" />
    </node>

<!-- CAMERA -->
   
   <node pkg="usb_cam" type="usb_cam_node" name="barc_cam"> 
        <param name="pixel_format" value="yuyv"/>
        <param name="video_device" value="/dev/video6"/>
        <param name="image_width" value="640"/>
        <param name="image_height" value="480"/>
		<!--<param name="brightness" value="50" />-->
		<!--<param name="contrast" value="60" />-->
   </node>
	<node name="image_proc" pkg="image_proc" type="image_proc" ns="barc_cam"/>


<!-- IMU -->
    <node pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
    </node>

<!-- ARDUINO -->
    <!--  * encoders and ultrasound sensors -->
    <node pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node>

<!-- OBSERVER-->
    <group ns="state_estimation">
    <!-- <node pkg="barc" type="ekf_Dyn.py" name="state_estimation"> -->
        <!-- v_x estimation sample time-->
        <param name="dt_v_enc" type="double" value="0.2" />
        <param name="v_x_min" type="double" value="0.2" />
        <!-- ekf properties -->
        <param name="q_std" type="double" value="0.01" />
        <param name="r_std_r" type="double" value="0.57" />
        <param name="r_std_v" type="double" value="1.2767" />
    </group>
    <!--</node>-->

		
<!-- LOW LEVEL CONTROLLER -->
    <node pkg="barc" type="controller_low_level.py" name="low_level_controller" output="screen"/>
</launch>
